.travis.yml

```diff
@@ -1,26 +1,44 @@
 language: python
+python:
+  - 2.7
+jdk:
+  - oraclejdk8
 
 install:
+  ### Java: We download now some dependencies to limit the number of messages later
+  - if $JAVA; then mvn dependency:resolve-plugins dependency:resolve -B -V ; fi
+  ### Python build dependencies
   - pip install tox coveralls
 script:
+  ### Java
+  - if $JAVA; then mvn verify -B -V ; fi
+  ### Python
   - tox
 after_success:
+  ### Python
   - coveralls
 
-# As of 2016-04 Travis provides Java 7 with precise so we use trusty
+### sudo is required by pip
 sudo: required
+
+### As of 2016-11, when Sudo is required, java8 is removed (WHY?)
 dist: trusty
+addons:
+  apt:
+    packages:
+      - oracle-java8-installer
+
 env:
   matrix:
-    - TOXENV=pylama
-    - TOXENV=pypy-coverage
-    - TOXENV=py27-coverage
+    - JAVA=true  TOXENV=pylama # We build Java once only
```

Comment:
Running the java tests in the pylama env is kind of weird.
What do you think about having a fourth case with JAVA=true and TOXENV being null ?
Also maybe we should make JAVA=false implicit.